invoice/invoice.py

```diff
@@ -47,6 +47,11 @@ def get_sorted_key(self, name):
             if purchase not in key:
                 key.append(purchase)
 
+        if self.origin and 'account.invoice.line' in str(self.origin):
+            invoice = self.origin.invoice
```

Comment:
Teòricament pot passar (i sempre acaba sortint algun cas extrany) que un camp de tipus Reference tingui un valor del tipus "account.invoice.line,-1" de manera que sí que té un string però no apunta a cap objecte en concret.

Crec que normalment el millor és fer:

if isinstance(self.origin, cls):
    invoice = self.origin.invoice
    if invoice not in key:
        key.append(invoice)

Reply:
Fet